### ISSUE TITLE
unary_union on a buffered empty Polygon causes Python Segmentation fault - GEOS 3.10

### ISSUE THREAD

**Issue by jlxu07517** [shapely/shapely#1230]
## Expected behavior and actual behavior.
Actual behavior (shapely 1.8.0) : `unary_union` on a buffered empty polygon causes python to crash with segmentation fault.
Expected behavior: Previous version of shapely 1.7.1 will return an empty polygon as expected.

## Steps to reproduce the problem.
```python
from shapely.ops import unary_union
from shapely.geometry import Polygon

unary_union(Polygon().buffer(0.1))
```
(For example, a script with required data)

## Operating system
Mac OS X 10.15.7

## Shapely version and provenance
shapely 0.8.1 from  conda-forge

**Comment by sgillies** [comment]
I can't reproduce this with a shapely 1.8.0 from PyPI, so I suspect it's a problem with the conda-forge build. There is a raft of new issues, all involving conda-forge.

**Comment by mwtoews** [comment]
I can reproduce it with GEOS 3.10.1 on Linux, so I suspect it's a GEOS issue. Here's what I did in a stock Python 3.6 from Ubuntu bionic:
```
$ export LD_LIBRARY_PATH=/path/to/GEOS/builds/geos-3.10.1/lib
$ gdb python
...
(gdb) run issue-1230.py
Starting program: /tmp/venv36/bin/python issue-1230.py
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[Detaching after fork from child process 3512]
[New Thread 0x7ffff1683700 (LWP 3513)]
[New Thread 0x7ffff0e82700 (LWP 3514)]
[New Thread 0x7fffe8681700 (LWP 3515)]
[New Thread 0x7fffd7e80700 (LWP 3516)]
[New Thread 0x7fffcf67f700 (LWP 3517)]
[New Thread 0x7fffc6e7e700 (LWP 3518)]
[New Thread 0x7fffbe67d700 (LWP 3519)]

Thread 1 "python" received signal SIGSEGV, Segmentation fault.
0x00007ffff4cc65b3 in geos::operation::geounion::CascadedPolygonUnion::binaryUnion(std::vector<geos::geom::Geometry const*, std::allocator<geos::geom::Geometry const*> > const&, unsigned long, unsigned long) () from /path/to/GEOS/builds/geos-3.10.1/lib/libgeos.so.3.10.1
```
so probably an upstream bug, which I'll follow-up....

**Comment by jorisvandenbossche** [comment]
> I can't reproduce this with a shapely 1.8.0 from PyPI, so I suspect it's a problem with the conda-forge build. We have a pair of new segfault reports, all involving conda-forge.

Just as a heads up that the latest packages from conda-forge started to use GEOS 3.10.0. So if there is a difference with the wheels, that's also a likely possible cause (as it seems to be in this case given @mwtoews comment), and not necessarily a build issue with conda-forge.

**Comment by mwtoews** [comment]
It's the same issue with GEOS 3.10.0 and 3.10.1, and nothing to do with conda-forge or building with CMake vs. autotools, as there is no error with a CMake GEOS 3.9.2 build. Not sure what the issue is yet.

**Comment by jorisvandenbossche** [comment]
And I can also reproduce the segfault (and also with pygeos).

It seems to boil down to the unary union of an empty polygon (the `Polygon()` in shapely creates an empty geometry collection, and buffering it turns it into an empty polygon):

```
In [2]: import pygeos

In [3]: pygeos.union_all([pygeos.Geometry("POLYGON EMPTY")])
Segmentation fault (core dumped)
```

**Comment by jorisvandenbossche** [comment]
Ah, I see you already simplified it to that as well in the upstream GEOS issue!

**Comment by jorisvandenbossche** [comment]
This was fixed in GEOS 3.10.2, and our packages now include a more recent GEOS version, so closing this issue.

**Comment by wboykinm** [comment]
@jorisvandenbossche @mwtoews @sgillies Big high fives to y'all for integrating the GEOS update here; it's been really helpful with our use of `unary_union`. 🙏
